renku/core/dataset/providers/dataverse.py

```diff
@@ -580,3 +582,12 @@ def make_file_url(file_id, base_url):
     args_dict = {"persistentId": file_id}
     url_parts[4] = urllib.parse.urlencode(args_dict)
     return urllib.parse.urlunparse(url_parts)
+
+
+class DataverseProviderPlugin(IDatasetProviderPlugin):
```

Comment:
Could we add this as a mixin on e.g. `DataverseProvider` directly?

Like 
```
class DataverseProvider(ProviderApi, IDatasetProviderPlugin):
[...]
    @hookimpl
    @classmethod  # not sure on the correct order of decorators
    def dataset_provider(cls) -> "Type[ProviderApi]":
        """The defintion of the provider."""
        return cls
[...]
```
Same for the other providers.

Reply:
Oh cool. A class method will do exactly what I was looking for. Don't know why I did not think of this.

I will also fix the mypy and style problems. Just wanted to see if this approach was acceptable or not.